tests/integration/metrics/test_integration_metric.py

```diff
@@ -0,0 +1,152 @@
+METRIC_MESSAGE_DATA = '{}'
+MOCKED_API_PORT = 9182
+
+MonkeyPatchFixture = NewType("MonkeyPatchFixture", Any)
+system_variables: Dict = {
+    'DYNATRACE_URL': 'http://localhost:' + str(MOCKED_API_PORT),
+    'REQUIRE_VALID_CERTIFICATE': 'False'
+    # 'DYNATRACE_ACCESS_KEY': ACCESS_KEY, this one is encoded in mocks files
+    # 'GCP_SERVICES': "gce_instance/default,api/default,gce_instance/default,cloudsql_database/default"
+}
+
+
+@pytest.fixture(scope="package", autouse=True)
+def setup_test_variables(resource_path_root):
+    lib.credentials._CLOUD_RESOURCE_MANAGER_ROOT = f"http://localhost:{MOCKED_API_PORT}/v1"
+    lib.credentials._SECRET_ROOT = f"http://localhost:{MOCKED_API_PORT}/v1"
+    lib.metric_ingest._MONITORING_ROOT = f"http://localhost:{MOCKED_API_PORT}/v3"
+    lib.entities.extractors.cloud_sql._SQL_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+    lib.entities.google_api._GCP_COMPUTE_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+    lib.entities.extractors.gce_instance._GCP_COMPUTE_ENDPOINT = f"http://localhost:{MOCKED_API_PORT}"
+
+    system_variables.update({
+        'GOOGLE_APPLICATION_CREDENTIALS': f"{resource_path_root}/metrics/token_for_tests.json",
+    })
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup_config(resource_path_root, mocker: MockerFixture):
+    # change the config dir
+    patched = mocker.patch('os.path.dirname')
+    patched.return_value = os.path.join(resource_path_root, "metrics")
+
+
+@pytest.fixture(scope="function", autouse=True)
+def setup_env(monkeypatch, resource_path_root):
+    for variable_name in system_variables:
+        monkeypatch.setenv(variable_name, system_variables[variable_name])
+
+
+@pytest.fixture(scope="function", autouse=True)
+def clean_journal():
+    Requests.reset_request_journal()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_wiremock():
+    # setup WireMock server
+
+    wiremock = None
+    Config.base_url = 'http://localhost:{}/__admin'.format(MOCKED_API_PORT)
+
+    wiremock = WireMockServer(port=MOCKED_API_PORT)
+    wiremock.start()
+
+    print("wiremock running")
+
+    # run test
+    yield
+
+    # stop WireMock server
+
+    wiremock.stop()
+
+
+@pytest.mark.asyncio
+async def test_metric_authorization_header():
+    await async_dynatrace_gcp_extension()
+
+    request = NearMissMatchPatternRequest(url_path_pattern="/api/v2/metrics/ingest",
+                                          method="POST")
+
+    matched_request: RequestResponseFindResponse = Requests.get_matching_requests(request)
+
+    assert_that(matched_request).is_not_empty()
+
+    result: RequestResponseRequest = matched_request.requests[0]
+    api_key = await get_dynatrace_secret_api_key()
+
+    assert_that(result.headers['Authorization']).is_equal_to(f"Api-Token {api_key}")
+
+
+async def get_dynatrace_secret_api_key():
+    context = LoggingContext(None)
+    async with init_gcp_client_session() as gcp_session:
```

Comment:
Why do we need this procedure if we have test value of the api key secret hard-coded in the mappings? It gave me a solid scare at first, because I thought this code is fetching real secret value 😄

Reply:
Right no need to, will change that.